nighthawk/test/statistic_test.cc

```diff
@@ -173,25 +179,21 @@ TYPED_TEST(TypedStatisticTest, ProtoOutput) {
 class StatisticTest : public testing::Test {};
 
 TEST(StatisticTest, HdrStatisticPercentilesProto) {
+  Envoy::Thread::ThreadFactory& thread_factory = Envoy::Thread::threadFactoryForTest();
+  Envoy::Stats::IsolatedStoreImpl store;
+  Envoy::Filesystem::InstanceImpl filesystem(100ms, thread_factory, store);
+  nighthawk::client::Statistic parsed_json_proto;
   HdrStatistic statistic;
-  int max = 10;
 
-  for (int i = 1; i <= max; i++) {
+  for (int i = 1; i <= 10; i++) {
     statistic.addValue(i);
   }
 
-  std::string str;
-  google::protobuf::util::JsonPrintOptions options;
-  google::protobuf::util::MessageToJsonString(statistic.toProto(), &str, options);
-  std::ifstream myfile;
-  // TODO(oschaaf): If this is confirmed to work in Google's environment, wrap
-  // Envoy::TestEnvironment::runfilesPath in a call within the Nighthawk namespace for re-use.
-  // TODO(oschaaf): Set up reusable helpers for .gold file testing, we'll probably get more of that.
-  // Also, move out the test data into its own directory.
-  myfile.open(Envoy::TestEnvironment::runfilesPath("nighthawk/test/hdr_proto_json.gold"));
-  std::string gold;
-  myfile >> gold;
-  EXPECT_EQ(gold, str);
+  Envoy::MessageUtil util;
+  util.loadFromJson(filesystem.fileReadToEnd(TestEnvironment::runfilesPath(
```

Comment:
Hmm, my suggested changes made this look worse, they were bad. I think we should have an explicit equality method, rather than implicit call on `MessageUtil`, but that change needs to happen upstream. Can you do this? I also didn't realize we needed a full filesystem song-and-dance, I thought we could just use a static method on the class. I'll merge this PR, but please do consider undoing my bad advice in a future PR.

Reply:
I'll take a look at upstreaming that change and incorporating that here.